src/App.js

```diff
@@ -29,37 +30,103 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  // const [winner, setWinner] = useState(null);
+  // const [gameEnd, setGameEnd] = useState(false);
 
-  // Wave 2
+  const squareClicked =(id)=>{
+    let didPlay = false 
+    if (winner !== null) {return}
+    const newState = squares.map(row =>row.map(pos =>{
+      if (pos.id !== id) {return pos;}
+      if (pos.value !== '') {return pos;} 
+  //we know the id matches
+      didPlay = true
+      const newPos = {...pos}
```

Comment:
👍 
Here, we spread the square position information from the existing state to copy it. When we set the new value, we only want to modify the new state data, not the existing state.